src/wp-includes/media.php

```diff
@@ -1091,9 +1091,17 @@ function wp_get_attachment_image( $attachment_id, $size = 'thumbnail', $icon = f
 		 */
 		$context = apply_filters( 'wp_get_attachment_image_context', 'wp_get_attachment_image' );
 
-		$attr           = wp_parse_args( $attr, $default_attr );
-		$attr['width']  = $width;
-		$attr['height'] = $height;
+		$attr = wp_parse_args( $attr, $default_attr );
+
+		// Ensure that the `$width` doesn't overwrite an already valid user-provided width.
+		if ( is_numeric( $width ) && ( ! isset( $attr['width'] ) || ! is_numeric( $attr['width'] ) ) ) {
```

Comment:
Let's check `isset` and `numeric` for `$attr['width']` then numeric for $width as it always have numeric value.

```suggestion
		if ( ( ! isset( $attr['width'] ) || ! is_numeric( $attr['width'] ) && is_numeric( $width ) ) ) {
```

Do same for height also.

Reply:
Thanks for the review. I found `$width` and `$height` can be `null` in two of our projects when we're debugging the missing `width/height` attributes the other day, only for SVG images.

But I think in this case, if I'm not wrong, there is no need to check `is_numeric( $width )` at all, checking `! isset( $attr['width'] ) || ! is_numeric( $attr['width'] )` alone should be enough here. I will take another closer look tomorrow. And I also wait and see if Adam has any further feedback.

Reply:
You're right about the `$width` and `$height` always have numeric value, I tracked down the codes to set the width and height.
https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/media.php#L480-L482

The reason that I got `null` for the `$width` and `$height` is due to a filter we're using internally in the projects at the company that I work at. So all good there, thanks again.